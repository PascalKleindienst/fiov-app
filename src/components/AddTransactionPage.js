// ==============================================
// ADD TRANSACTION PAGE COMPONENT ===============
// ==============================================
import React from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';
import { getTranslate } from 'react-localize-redux';
import TransactionForm from './TransactionForm';
import { addTransaction } from '../actions/transactions';

export class AddTransactionPage extends React.Component {
    onSubmit = (transaction) => {
        this.props.addTransaction(transaction);
        this.props.history.push('/');
    }

    render() {
        return (
            <div>
                <div className="page-header">
                    <div className="content-container">
                        <h2 className="page-header__title">{ this.props.translate('transactions.add_transaction') }</h2>
                    </div>
                </div>
                <div className="content-container">
                    <TransactionForm onSubmit={this.onSubmit} />
                </div>
            </div>
        );
    }
}

// PropTypes
AddTransactionPage.propTypes = {
    translate: PropTypes.func.isRequired,
    addTransaction: PropTypes.func.isRequired
};

// States
const mapStateToProps = (state) => ({
    translate: getTranslate(state.locale)
});

// Dispatch
const mapDispatchToProps = (dispatch) => ({
    addTransaction: (transaction) => dispatch(addTransaction(transaction))
});

export default connect(mapStateToProps, mapDispatchToProps)(AddTransactionPage);