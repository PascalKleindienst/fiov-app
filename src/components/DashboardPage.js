// ==============================================
// DASHBOARD PAGE COMPONENT =====================
// ==============================================
import React from 'react';
import TransactionList from './TransactionList';
import TransactionListFilters from './TransactionListFilters';

const DashboardPage = () => (
    <div>
        <TransactionListFilters />
        <TransactionList />
    </div>
);

export default DashboardPage;
