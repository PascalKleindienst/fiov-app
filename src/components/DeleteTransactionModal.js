// ==============================================
// DELETE TRANSACTION MODAL COMPONENT ===========
// ==============================================
import React from 'react';
import PropTypes from 'prop-types';
import Modal from 'react-modal';
import { localize } from 'react-localize-redux';

export const DeleteTransactionModal = (props) => (
    <Modal
        isOpen={!!props.isOpen}
        contentLabel={ props.translate('transactions.modal.delete') }
        onRequestClose={props.handleClose}
        ariaHideApp={false}
        closeTimeoutMS={200}
        className="modal"
        style={{ overlay: {backgroundColor: 'rgba(0,0,0,.75)' }}}
    >
        <h3 className="modal__title">{ props.translate('transactions.modal.delete') }</h3>
        <p className="modal__body">{ props.translate('transactions.modal.confirm') }</p>
        <div className="modal__actions">
            <button className="button" onClick={props.handleConfirmation}>{ props.translate('transactions.modal.delete') }</button>
            <button className="button button--secondary" onClick={props.handleClose}>{ props.translate('cancel') }</button>
        </div>
    </Modal>
);

// PropTypes
DeleteTransactionModal.propTypes = {
    translate: PropTypes.func.isRequired,
    isOpen: PropTypes.bool.isRequired,
    handleClose: PropTypes.func.isRequired,
    handleConfirmation: PropTypes.func.isRequired
};

export default localize(DeleteTransactionModal, 'locale');