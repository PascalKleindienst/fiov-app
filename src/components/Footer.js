// ==============================================
// HOME PAGE COMPONENT ========)=================
// ==============================================
import React from 'react';
import PropTypes from 'prop-types';
import { Link } from 'react-router-dom';
import { localize } from 'react-localize-redux';

export const smoothscroll = () => {
    const currentScroll = document.documentElement.scrollTop || document.body.scrollTop;
    
    if (currentScroll > 0) {
        window.requestAnimationFrame(smoothscroll);
        window.scrollTo(0,currentScroll - (currentScroll/5));
    }
};

export const Footer = ({ translate }) => (
    <footer className="footer">
        <div className="content-container">
            <div className="footer__content">
                <Link to="/imprint">{ translate('menu.imprint') }</Link>
                <Link to="/privacy">{ translate('menu.privacy') }</Link>
                <span className="scroll-top"  onClick={smoothscroll}></span>
            </div>
        </div>
    </footer>
);

// PropTypes
Footer.propTypes = {
    translate: PropTypes.func.isRequired
};

export default localize(Footer, 'locale');
