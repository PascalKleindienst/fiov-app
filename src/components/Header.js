// ==============================================
// HEADER COMPONENT==============================
// ==============================================
import React from 'react';
import PropTypes from 'prop-types';
import { Link, NavLink } from 'react-router-dom';
import { connect } from 'react-redux';
import { getTranslate } from 'react-localize-redux';
import { logout } from '../actions/auth';

// Header Component
export const Header = ({ auth, isAuthenticated, logout, translate }) => (
    <header className="header">
        <div className="content-container">
            { isAuthenticated ? (
                <div className="header__content">
                    <Link className="header__title" to="/dashboard">
                        <h1>{ translate('title') }</h1>
                    </Link>
                    <div className="header__nav">
                        <NavLink to="/add" className="header__nav__item" activeClassName="header__nav__item--active">
                            { translate('transactions.add_transaction') }
                        </NavLink>
                        
                        <div className="profile has-dropdown">
                            <img src={ auth.picture } alt="" className="profile__picture" />
                            <div className="dropdown-menu">
                                <div className="dropdown-menu__header">
                                    <strong>{ translate('user.account') }</strong>
                                </div>
                                <Link className="dropdown-menu__item" to="/profile">
                                    { translate('user.profile') }
                                </Link>
                                <Link className="dropdown-menu__item logout" onClick={logout} to="#">
                                    { translate('user.logout') }
                                </Link>
                            </div>
                        </div>
                    </div>
                </div>
            ) : (
                <div className="header__content">
                    <Link className="header__title" to="/">
                        <h1>{ translate('title') }</h1>
                    </Link>
                    <div className="header__nav">
                        <Link to="/login" className="header__nav__item">{ translate('menu.login') }</Link>
                    </div>
                </div>
            )}
        </div>
    </header>
);

// PropTypes
Header.propTypes = {
    auth: PropTypes.shape({
        picture: PropTypes.string,
    }),
    isAuthenticated: PropTypes.bool.isRequired,
    logout: PropTypes.func.isRequired,
    translate: PropTypes.func.isRequired
};

// States
const mapStateToProps = (state) => ({
    auth: state.auth,
    isAuthenticated: !!state.auth.id,
    translate: getTranslate(state.locale)
});

// Dispatch Functions
const mapDispatchToProps = (dispatch) => ({
    logout: () => dispatch(logout())
});

export default connect(mapStateToProps, mapDispatchToProps)(Header);
