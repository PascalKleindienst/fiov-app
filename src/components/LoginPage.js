// ==============================================
// LOGIN PAGE COMPONENT =========================
// ==============================================
import React from 'react';
import PropTypes from 'prop-types';
import { localize } from 'react-localize-redux';

export const LoginPage = ({ translate }) => (
    <div className="box-layout">
        <div className="box-layout__box">
            <h1 className="box-layout__title">{ translate('title') }</h1>
            <p>{ translate('subtitle') }</p>
            <p>
                <a className="button" href="/auth/twitter">{ translate('twitterLogin') }</a>
            </p>
            <p>
                <a className="button" href="/auth/facebook">{ translate('facebookLogin') }</a>
            </p>
            <p>
                <a className="button" href="/auth/google">{ translate('googleLogin') }</a>
            </p>
        </div>
    </div>
);

// PropTypes
LoginPage.propTypes = {
    translate: PropTypes.func.isRequired
};

export default localize(LoginPage, 'locale');
