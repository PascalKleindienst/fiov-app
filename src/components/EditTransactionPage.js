// ==============================================
// EDIT TRANSACTION PAGE COMPONENT ==============
// ==============================================
import React from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';
import { getTranslate } from 'react-localize-redux';
import TransactionForm from './TransactionForm';
import { editTransaction, removeTransaction } from '../actions/transactions';
import DeleteTransactionModal from './DeleteTransactionModal';

export class EditTransactionPage extends React.Component {
    /**
     * Sets showModal state to false
     */
    constructor () {
        super();
        this.state = {
            showModal: false
        };
    }

    /**
     * Open Modal
     */
    handleOpenModal = () => {
        this.setState({ showModal: true });
    }

    /**
     * Close Modal
     */
    handleCloseModal = () => {
        this.setState({ showModal: false });
    }

    /**
     * Edit Transaction
     */
    onSubmit = (transaction) => {
        this.props.editTransaction(this.props.transaction.id, transaction);
        this.props.history.push('/');
    }

    /**
     * Remove Transaction
     */
    onRemove = () => {
        this.props.removeTransaction({ id: this.props.transaction.id });
        this.props.history.push('/');
    }

    render() {
        return (
            <div>
                <div className="page-header">
                    <div className="content-container">
                        <h2 className="page-header__title">{ this.props.translate('transactions.edit_transaction') }</h2>
                    </div>
                </div>
                <div className="content-container">
                    <TransactionForm
                        transaction={this.props.transaction}
                        onSubmit={this.onSubmit}
                    />

                    <button className="button button--secondary button--remove" onClick={this.handleOpenModal}>{ this.props.translate('transactions.remove_transaction') }</button>
                </div>
                <DeleteTransactionModal
                    isOpen={this.state.showModal}
                    handleClose={this.handleCloseModal}
                    handleConfirmation={this.onRemove}
                />
            </div>
        );
    }
};

// PropTypes
EditTransactionPage.propTypes = {
    translate: PropTypes.func.isRequired,
    history: PropTypes.object.isRequired,
    transaction: PropTypes.object.isRequired,
    editTransaction: PropTypes.func.isRequired,
    removeTransaction: PropTypes.func.isRequired
};

// States
const mapStateToProps = (state, props) => ({
    translate: getTranslate(state.locale),
    transaction: state.transactions.find((transaction) => transaction.id == props.match.params.id)
});

const mapDispatchToProps = (dispatch, props) => ({
    editTransaction: (id, transaction) => dispatch(editTransaction(id, transaction)),
    removeTransaction: (data) => dispatch(removeTransaction(data))
});

export default connect(mapStateToProps, mapDispatchToProps)(EditTransactionPage);