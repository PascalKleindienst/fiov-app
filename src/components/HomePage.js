// ==============================================
// HOME PAGE COMPONENT ========)=================
// ==============================================
import React from 'react';
import PropTypes from 'prop-types';
import { Link } from 'react-router-dom';
import { localize } from 'react-localize-redux';
import Header from './Header';
import Footer from './Footer';

const HomePage = ({ translate }) => (
    <div id="home-page">
        <Header />
        <section className="content">
            <div className="jumbotron">
                <img className="" style={{ maxWidth: '100%' }} src="/images/preview.png" />
            </div>
            <div className="content-container">
                <p>lorem ipsum dolor sit amet, lorem ipsum dolor sit amet lorem ipsum dolor sit amet lorem ipsum dolor sit amet lorem. lorem ipsum dolor sit amet, lorem ipsum dolor sit amet lorem ipsum dolor sit amet lorem ipsum dolor sit amet lorem.  lorem ipsum dolor sit amet, lorem ipsum dolor sit amet lorem ipsum dolor sit amet lorem ipsum dolor sit amet lorem. ipsum dolor sit amet lorem ipsum dolor sit amet, lorem ipsum dolor sit amet lorem ipsum dolor sit amet lorem ipsum dolor sit ametlorem ipsum dolor sit amet </p>
                <p>lorem ipsum dolor sit amet, lorem ipsum dolor sit amet lorem ipsum dolor sit amet lorem ipsum dolor sit amet lorem ipsum dolor sit amet lorem ipsum dolor sit amet, lorem ipsum dolor sit amet lorem ipsum dolor sit amet lorem ipsum dolor sit ametlorem ipsum dolor sit amet </p>
                <p>lorem ipsum dolor sit amet, lorem ipsum dolor sit amet lorem ipsum dolor sit amet lorem ipsum dolor sit amet lorem ipsum dolor sit amet lorem ipsum dolor sit amet, lorem ipsum dolor sit amet lorem ipsum dolor sit amet lorem ipsum dolor sit ametlorem ipsum dolor sit amet. lorem ipsum dolor sit amet, lorem ipsum dolor sit amet lorem ipsum dolor sit amet lorem ipsum dolor sit amet lorem. lorem ipsum dolor sit amet, lorem ipsum dolor sit amet lorem ipsum dolor sit amet lorem ipsum dolor sit amet lorem. lorem ipsum dolor sit amet, lorem ipsum dolor sit amet lorem ipsum dolor sit amet lorem ipsum dolor sit amet lorem. lorem ipsum dolor sit amet, lorem ipsum dolor sit amet lorem ipsum dolor sit amet lorem ipsum dolor sit amet lorem.  </p>
                <p>lorem ipsum dolor sit amet, lorem ipsum dolor sit amet lorem ipsum dolor sit amet lorem ipsum dolor sit amet lorem ipsum dolor sit amet lorem ipsum dolor sit amet, lorem ipsum dolor sit amet lorem ipsum dolor sit amet lorem ipsum dolor sit ametlorem ipsum dolor sit amet </p>
            </div>
        </section>
        <Footer />
    </div>
);

// PropTypes
HomePage.propTypes = {
    translate: PropTypes.func.isRequired
};

export default localize(HomePage, 'locale');
