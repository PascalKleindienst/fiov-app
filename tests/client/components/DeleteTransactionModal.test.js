// ==============================================
// DELETETRANSACTIONMODAL PAGE TEST ================
// ==============================================
import React from 'react';
import { shallow } from 'enzyme';
import { DeleteTransactionModal } from '../../../src/components/DeleteTransactionModal';

describe('my delete transaction model component', () => {
    test('should render correctly', () => {
        const props = {
            translate: (msg) => msg,
            isOpen: true,
            handleClose: () => false,
            handleConfirmation: () => false
        }
        const wrapper = shallow(<DeleteTransactionModal {...props} />);
        expect(wrapper).toMatchSnapshot();
    });
});