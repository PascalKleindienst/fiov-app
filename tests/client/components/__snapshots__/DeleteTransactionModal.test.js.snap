// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`my delete transaction model component should render correctly 1`] = `
<Modal
  ariaHideApp={false}
  bodyOpenClassName="ReactModal__Body--open"
  className="modal"
  closeTimeoutMS={200}
  contentLabel="transactions.modal.delete"
  isOpen={true}
  onRequestClose={[Function]}
  parentSelector={[Function]}
  portalClassName="ReactModalPortal"
  shouldCloseOnEsc={true}
  shouldCloseOnOverlayClick={true}
  shouldFocusAfterRender={true}
  shouldReturnFocusAfterClose={true}
  style={
    Object {
      "overlay": Object {
        "backgroundColor": "rgba(0,0,0,.75)",
      },
    }
  }
>
  <h3
    className="modal__title"
  >
    transactions.modal.delete
  </h3>
  <p
    className="modal__body"
  >
    transactions.modal.confirm
  </p>
  <div
    className="modal__actions"
  >
    <button
      className="button"
      onClick={[Function]}
    >
      transactions.modal.delete
    </button>
    <button
      className="button button--secondary"
      onClick={[Function]}
    >
      cancel
    </button>
  </div>
</Modal>
`;
